cupy/core/_cub_reduction.pyx

```diff
@@ -263,6 +265,10 @@ cpdef inline tuple _can_use_cub_block_reduction(
 
     # check reduction axes, if not contiguous then fall back to old kernel
     if in_arr._f_contiguous:
+        order = 'f'
+        if not cub._cub_device_segmented_reduce_axis_compatible(
+                reduce_axis, in_arr.ndim, order):
+            return None
```

Comment:
Looks like the correct thing to do, thanks Gregory! Perhaps we should just call `_cub_device_segmented_reduce_axis_compatible()` for both C and F orders, and then construct `axis_permutes_cub`?

Reply:
I have now added this call to the C-contiguous case as well

Reply:
?? Did you push @grlee77?

Reply:
whoops. I did it, but forgot to commit the change. It is there now in f83a0ca